plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -191,8 +191,12 @@ internal suspend fun mapSnippetFindings(
             snippetLicenseFindings
         )
 
-        runningSnippetCount += mappedSnippets.size
-        results += mappedSnippets
+        val snippetFindingIterator = mappedSnippets.iterator()
+        while (runningSnippetCount < snippetsLimit && snippetFindingIterator.hasNext()) {
+            val snippetFinding = snippetFindingIterator.next()
+            runningSnippetCount += snippetFinding.snippets.size
+            results += snippetFinding
+        }
```

Comment:
Maybe instead something like

        results += mappedSnippets.takeWhile {
            runningSnippetCount += it.snippets.size
            runningSnippetCount < snippetsLimit
        }

Reply:
And BTW, is the condition in line 182 then still needed / correct?

Reply:
Your suggestion with `takeWhile` does not work because the condition is evaluated _before_ adding the element to the list. Therefore, when you are going over the limit (condition == true) you are not adding the current element, while in the _iterator_ logic it is added.

And yes the condition line 182 is still needed as it allows to stop loading snippets from FossID when the limit has been reached.